tests/ecosystem/upgrade/test_noobaa.py

```diff
@@ -159,3 +169,48 @@ def test_buckets_after_upgrade(upgrade_buckets, mcg_obj_session):
     """
     for bucket in mcg_obj_session.read_system().get('buckets'):
         assert bucket.get('mode') == BS_OPTIMAL
+
+
+@pre_upgrade
+def test_start_upgrade_mcg_io(mcg_workload_job):
+    """
+    Confirm that there is MCG workload job running before upgrade.
+    """
+    job_status = None
+    job_name = mcg_workload_job.ocp.resource_name
+    # wait a few seconds for fio job to start
+    logger.info(f"Checking number of active pods for job {job_name}")
+    for i in range(0, 5):
```

Comment:
plz consider use TimeoutSampler instead this loop

Reply:
+1

Reply:
Thanks for the tip. I considered it but I would rather keep it as it is. There would have to be created special function inside test function and that would be passed to TimeoutSampler which would be in my opinion less readable. Also there would have to be a new call to the new function in `assert job_status == 1` in assert message which would lead to reporting wrong number of pods. It would look something like:
```
     assert TimeoutSampler(                                                                                                                                                                                                                   
         timeout=20, sleep=3, func=new_function, state=1                                                                                                                                                      
     ), (                                                                                                                                                                                                                                     
         f"Number of running pods for job {job_name} is not 1, "                                                                                                                                                                              
         f"it is {new_function()}"                                                                                                                                                                                                          
     )    
```

Reply:
I tend to agree with Filip's point - but considering that lines 189 + 193-196, and 212-16 are outright code duplication - I'd reconsider moving the logic into a helper function that the two tests can later use.
Also, it's possible to give up on the one-liner `assert TimeoutSampler`, and instead save the return value on the side
i.e. - 
```
def check_job_state(job, desired_state):
    job_name = job.ocp.resource_name
    job_obj = job.ocp.get(resource_name=job_name, out_yaml_format=True)
    def _retrieve_job_state():
        return job.get('items').get(0).get('status').get('active')
    try:
        for state in TimeoutSampler(timeout=20, sleep=3, func=_retrieve_job_state):
            if state == desired_state:
                return True
    except TimeoutExpiredError:
        log.error(
            f"Number of running pods for job {job_name} is not 1, "
        ) 
        return False
```
Pretty much turns both of the tests into one-liners

Reply:
Ok, I changed it based on Ben's comment. The code duplication is a good point.